includes/wp-cli.php

```diff
@@ -8,96 +8,152 @@
+		$formatted_records = array();
 
-		$start      = microtime( true );
-		$query_args = array();
+		$this->connection();
 
-		foreach ( $assoc_args as $key => $value ) {
-			$query_args[ $key ] = $value;
+		if ( empty( $assoc_args['fields'] ) ) {
+			$fields = array( 'created', 'ip', 'author', 'author_meta.user_login', 'author_role', 'summary' );
+		} else {
+			$fields = $assoc_args['fields'];
 		}
 
-		if ( empty( $query_args['fields'] ) ) {
-			$defaults = array( 'created', 'ip', 'author', 'author_meta.user_login', 'author_role', 'summary' );
-
-			/**
-			 * Filter for default fields when arg is not provided
-			 *
-			 * @return array $defaults
-			 */
-			$defaults = apply_filters( 'wp_stream_wp_cli_default_fields', (array) $defaults );
-
-			$query_args['fields'] = implode( ',', array_map( 'trim', $defaults ) );
-		}
+		$query_args['fields'] = implode( ',', $fields );
 
 		$records = wp_stream_query( $query_args );
 
-		if ( empty( $records ) ) {
-			WP_CLI::success( __( 'No results found.', 'stream' ) );
-			return;
+		// make structure Formatter compatible
+		foreach ( $records as $key => $record ) {
+			$formatted_records[$key] = array();
+			foreach ( $record as $field_name => $field ) {
+				$formatted_records[$key] = array_merge( $formatted_records[$key], $this->any2array( $field_name, $field ) );
+			}
 		}
 
-		$fields = array_map( 'trim', explode( ',', $query_args['fields'] ) );
-
-		foreach ( $records as $record ) {
-			$output = '';
-
-			foreach ( $fields as $field ) {
-				$values = wp_list_pluck( $records, $field );
-
-				array_walk( $values, function( &$value ) {
-					$value = is_array( $value ) ? $value[0] : $value;
-				});
-
-				$longest = max( array_map( 'strlen', $values ) );
-				$value   = is_array( $record->$field ) ? implode( ', ', $record->$field ) : $record->$field;
-				$output .= $value;
-				$diff    = absint( $longest - strlen( $value ) );
+		$formatter = new \WP_CLI\Formatter(
+			$assoc_args,
+			$fields
+		);
+		$formatter->display_items( $formatted_records );
+	}
 
-				for ( $i = 0; $i < $diff; $i++ ) {
-					$output .= ' ';
-				}
+	/**
+	 * Convert any value to a flat array.
+	 *
+	 * @param string $name The output array element name.
+	 * @param mixed $object Any value to be converted to an array.
+	 * @return array The flat array.
+	 */
+	private function any2array( $name, $object ) {
+		$array = array();
 
-				$output .= '   ';
+		if ( is_object( $object ) ) {
+			foreach ( $object as $key => $property ) {
+				$array = array_merge( $array, $this->any2array( $name . '.' . $key, $property ) );
 			}
-
-			WP_CLI::line( trim( $output ) );
+		} elseif ( is_array( $object ) ) {
+			$array[$name] = $object[0];
+		} else {
+			$array[$name] = $object;
 		}
-
-		$found   = count( $records );
-		$stop    = microtime( true ) - $start;
-		$message = sprintf( _n( '1 result found in %s seconds.', '%s results found in %s seconds.', $found, 'stream' ), number_format( $found ), number_format( $stop, 4 ) );
```

Comment:
I added the results count back in, it's helpful meta to know when dealing with large result sets.

https://github.com/wp-stream/stream/commit/1377185f004fac5486fba3b749488889be948312